engine/src/desktop-menu.cpp

```diff
@@ -444,7 +444,8 @@ void MCButton::macopenmenu(void)
                 // SN-2014-08-25: [[ Bug 13240 ]] We need to keep the actual popup_menustring,
                 //  in case some menus are nested
                 MCStringRef t_menupick;
-                t_menupick = s_popup_menupick;
+                // PM-2015-02-09: [[ Bug 14521 ]] Make sure we don't pass nil to message_with_valueref_args
+                t_menupick = !MCStringIsEmpty(s_popup_menupick) ? s_popup_menupick : kMCEmptyString;
```

Comment:
In what case is s_popup_menupick nil? Surely if the popup is not cancelled, MCPlatformPopUp returns true and therefore there must be something non-nil like in s_popup_menupick...